packages/ui/src/components/va-counter/VaCounter.vue

```diff
@@ -0,0 +1,391 @@
+    }
+
+    const isMinReached = computed(() => {
+      if (!props.min) { return false }
+      return +valueComputed.value === props.min
+    })
+
+    const isMaxReached = computed(() => {
+      if (!props.max) { return false }
+      return +valueComputed.value > (props.max - props.step)
+    })
+
+    const disabledDecreaseAction = computed(() => (
+      isMinReached.value || props.readonly || props.disabled
+    ))
+
+    const disabledIncreaseAction = computed(() => (
+      isMaxReached.value || props.readonly || props.disabled
+    ))
+
+    const decreaseCount = () => {
+      if (disabledDecreaseAction.value) { return }
+      calculateCounterValue(+valueComputed.value - props.step)
+    }
+
+    const increaseCount = () => {
+      if (disabledIncreaseAction.value) { return }
+      calculateCounterValue(+valueComputed.value + props.step)
+    }
+
+    const { colorComputed } = useColor(props)
+
+    const decreaseIconProps = computed(() => ({
+      class: { 'va-counter__icon--inactive': disabledDecreaseAction.value },
+      color: colorComputed.value,
+      name: props.decreaseIcon,
+      onClick: !disabledDecreaseAction.value ? decreaseCount : null,
+    }))
+
+    const increaseIconProps = computed(() => ({
+      class: { 'va-counter__icon--inactive': disabledIncreaseAction.value },
+      color: colorComputed.value,
+      name: props.increaseIcon,
+      onClick: !disabledIncreaseAction.value ? increaseCount : null,
+    }))
+
+    const squareCorners = computed(() => (
+      (typeof props.margins === 'string' ? parseFloat(props.margins) : props.margins) === 0
+    ))
+
+    const decreaseButtonProps = computed(() => ({
+      ...pick(props, ['outline', 'rounded', 'color', 'textColor']),
+      flat: props.flat && !props.outline,
+      outline: props.flat && props.outline,
+      disabled: disabledDecreaseAction.value,
+    }))
+
+    const increaseButtonProps = computed(() => ({
+      ...pick(props, ['rounded', 'color', 'textColor']),
+      flat: props.flat && !props.outline,
+      outline: props.flat && props.outline,
+      disabled: disabledIncreaseAction.value,
+    }))
+
+    const computedInputAttributes = computed(() => ({
+      ariaLabel: props.label,
+      ...omit(attrs, ['class', 'style']),
+      ...pick(props, ['disabled', 'min', 'max', 'step']),
+      readonly: props.readonly || !props.manualInput,
+    }) as InputHTMLAttributes)
+
+    const computedClass = computed(() => ([
+      attrs.class,
+      { 'va-counter--input-square': squareCorners.value },
+    ]))
+
+    const computedStyle: ComputedRef<Partial<CSSStyleDeclaration>> = computed(() => {
+      const style = attrs.style || {}
+      return Object.assign(style, {
+        width: typeof props.width === 'number' ? `${props.width}px` : props.width,
+      })
+    })
+
+    const computedMargins = computed(() => (
+      typeof props.margins === 'number' ? `${props.margins}px` : props.margins
+    ))
```

Comment:
It is a computedMargin (left and right), not margins.

PX can be changed with FN, same for width. It can be a function from utils folder.

```suggestion
    const computedMargin = computed(() => safeCSSLength(props.margins))
```

Reply:
Done